src/core/TriangleSplitter.js

```diff
@@ -207,8 +260,9 @@ export class TriangleSplitter {
 				}
 
 				// we only don't consider this an intersection if the start points hits the plane
-				if ( Math.abs( startDist ) < COPLANAR_EPSILON ) {
+				if ( Math.abs( startDist ) < coPlanarEpsilonStart ) {
 
+					plane.projectPoint( _edge.start, arr[ t ] );
```

Comment:
Can you explain this addition?

Reply:
If we estimate that _edge.start belongs to plane, then we make sure _edge.start is on the plane to lower the uncertainty.
It is to avoid error propagation after multiple CSG operations on the same geometry.

Reply:
The point of this condition block, though, is to _skip_ interpreting the start point as an intersection if it's the only point that lies on the plane. This is so we avoid double counting the same vertex if the plane passes through a corner of the triangle.

I'm also seeing now that this is modifying the same vector that is on the triangle object (ie this line could be changing `tri.a`). Same with the other point projections above. Is this intentional? My feeling is that "tri" should be treated as immutable in this context so we're not changing triangle vertices over the course of the function.